src/config.rs

```diff
@@ -2690,21 +2669,86 @@ impl Default for File {
     }
 }
 
-#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, OnlineConfig)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct LogConfig {
-    #[serde(with = "log_level_serde")]
-    pub level: slog::Level,
+    pub level: LogLevel,
+    #[online_config(skip)]
     pub format: LogFormat,
+    #[online_config(skip)]
     pub enable_timestamp: bool,
+    #[online_config(skip)]
     pub file: File,
 }
 
+/// LogLevel is a wrapper type of `slog::Level`
```

Comment:
Why?

Reply:
to implement some traits like From<ConfigValue>

Reply:
Maybe we can implement a new attribue like serde's `with=..` to avoid this kind of boilerplate

Reply:
Only `impl From<LogLevel> for ConfigValue` is required? `impl TryFrom<ConfigValue> for LogLevel` seems to be optional (can be implemented as a function).

Is it possible to avoid it by blanket implement this:
```
impl From<T> for ConfigValue where T: std::string::ToString
```

Reply:
`std::string::ToString` is implemented for every type that implements `std::fmt::Display` implicitly. So we may unexpected implement `ConfigValue` for non-string types here.

See: https://github.com/rust-lang/rust/blob/4603ac31b0655793a82f110f544dc1c6abc57bb7/library/alloc/src/string.rs#L2479-L2493

Reply:
Could you give an example case where we don't want it to use `std::fmt::Display`?

Reply:
e.g ReadableDuration, it implement `Display` but has its own ConfigValue variant `ConfigValue::Duration`, See:
https://github.com/tikv/tikv/blob/0dff1be50281c72c55c5464751cd733032115ce4/components/tikv_util/src/config.rs#L439

Reply:
I see.